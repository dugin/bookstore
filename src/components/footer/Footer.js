import React from 'react';


const Footer = () => {

    const styles = {
        textAlign: 'center',
        marginTop: '20px'
    };
    return (
        <div style={styles}>
            <p>Developed by Rodrigo Dugin</p>
        </div>
    );
}

export default Footer;
