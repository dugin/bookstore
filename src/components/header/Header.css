.Header {
  justify-content: center;
  display: flex;
  align-items: center;
  min-height: 65px;
  box-shadow: 12px 15px 20px 0 rgba(46, 61, 73, 0.15);
  background-color: #3fc35f; }
  .Header__title {
    color: #fff; }
    .Header__title:hover {
      color: #fff; }
  .Header__links {
    position: absolute;
    right: 10px; }
    .Header__links__icon {
      font-size: 32px;
      color: #fff; }
      .Header__links__icon:hover {
        cursor: pointer; }
