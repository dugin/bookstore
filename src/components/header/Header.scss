@import '../../variables';

.Header {
  justify-content: center;
  display: flex;
  align-items: center;
  min-height: 65px;
  box-shadow: 12px 15px 20px 0 rgba(46, 61, 73, 0.15);
  background-color: $color__primary;

  &__title {
    color: $color__text--white;

    &:hover{
      color: $color__text--white;
    }
  }

  &__links {
    position: absolute;
    right: 10px;

    &__icon {
      font-size: 32px;
      color: $color__text--white;

      &:hover{
        cursor: pointer;
      }

    }
  }
}