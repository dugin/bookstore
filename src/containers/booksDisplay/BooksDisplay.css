.BooksDisplay {
  min-height: calc(100vh - 140px);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center; }
  .BooksDisplay__loading {
    color: #c33fa3; }
