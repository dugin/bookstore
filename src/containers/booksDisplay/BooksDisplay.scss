@import '../../variables';

.BooksDisplay {
  min-height: calc(100vh - 140px);
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  &__loading{
    color: $color--accent;

  }
}