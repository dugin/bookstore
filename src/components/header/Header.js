import React from 'react';
import {Collapse, Navbar, NavbarToggler, NavbarBrand, Nav, NavItem, NavLink} from 'reactstrap';
import './Header.css'

const Header = (props) => {

    return (
        <header className="Header">
            <NavbarBrand className="Header__title" href="/">Bookstore</NavbarBrand>
            <Nav className="Header__links">
                <NavItem>
                    <NavLink><i className="Header__links__icon material-icons">shopping_cart</i></NavLink>
                </NavItem>
            </Nav>
        </header>
    )

};
export default Header;