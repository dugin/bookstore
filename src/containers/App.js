import React, {Component} from 'react';
import './App.scss';
import Header from "../components/header/Header";
import Footer from "../components/footer/Footer";
import BooksDisplay from "./booksDisplay/BooksDisplay";

class App extends Component {
    render() {
        return (
            <div className="App">
                <Header/>
                <BooksDisplay/>
                <Footer/>
            </div>
        );
    }
}

export default App;
