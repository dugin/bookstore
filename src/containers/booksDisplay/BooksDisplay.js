import React, {Component} from 'react';
import Book from "../book/Book";
import * as BooksAPI from '../../api/BooksAPI'
import './BooksDisplay.css'
import ReactLoading from 'react-loading';

class BooksDisplay extends Component {

    static books;

    constructor(props) {
        super(props);
        this.state = {books: [], isLoading: true};
    }

    componentDidMount() {
        this.getBooks();
    }

    getBooks() {
        this.setState({isLoading: true});

        BooksDisplay.books ? this.setBooks(BooksDisplay.books) :
            BooksAPI.get()
                .then(books => {
                    this.setBooks(books);
                })
    }

    setBooks(books) {
        BooksDisplay.books = books;
        this.setState({books, isLoading: false})
    }

    render() {
        return (
            <div className="BooksDisplay container px-0 mt-3">
                {this.state.isLoading ? (
                    <div className="BooksDisplay__loading">
                        <ReactLoading className="mb-4" type="spinningBubbles" delay={0} color="#c33fa3" height={75} width={75}/>
                        Carregando...</div>
                ) : (
                    <div className="row no-gutters">
                        {this.state.books
                            .map(book => (
                                <div key={book.id}
                                     className="BooksDisplay__bookWrapper col-12 col-sm-12 col-md-6 col-lg-4  col-xl-3 ">
                                    <Book book={book}/>
                                </div>
                            ))}
                    </div>
                )}

            </div>
        );
    }
}

export default BooksDisplay;