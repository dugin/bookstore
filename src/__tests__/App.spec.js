import React from 'react';
import App from '../containers/App';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from "../components/header/Header";
import Footer from "../components/footer/Footer";
import BooksDisplay from "../containers/booksDisplay/BooksDisplay";


describe('App', () => {
    const app = shallow(<App/>);

    it('renders properly', () => {
        expect(toJson(app)).toMatchSnapshot();
    });

    it('contains `Header`', () => {
        expect(shallow(<Header/>).exists()).toBe(true);
    });

    it('contains `BooksDisplay`', () => {
        expect(shallow(<BooksDisplay/>).exists()).toBe(true);
    });

    it('contains `Footer`', () => {
        expect(shallow(<Footer/>).exists()).toBe(true);
    });
})