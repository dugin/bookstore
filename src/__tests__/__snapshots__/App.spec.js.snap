// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<div
  className="App"
>
  <Header />
  <BooksDisplay />
  <Footer />
</div>
`;
